sensu/plugins/memory-metrics.rb

```diff
@@ -13,26 +13,27 @@ class MemoryGraphite < Sensu::Plugin::Metric::CLI::Graphite
 
   def run
     # Metrics borrowed from hoardd: https://github.com/coredump/hoardd
-
     mem = metrics_hash
-
     mem.each do |k, v|
-      output "#{config[:scheme]}.#{k}", v
+      if k and v
+        output "#{config[:scheme]}.#{k}", v
+      end
     end
-
     ok
   end
 
   def metrics_hash
     mem = {}
     meminfo_output.each_line do |line|
-      mem['total']     = line.split(/\s+/)[1].to_i * 1024 if line.match(/^MemTotal/)
-      mem['free']      = line.split(/\s+/)[1].to_i * 1024 if line.match(/^MemFree/)
-      mem['buffers']   = line.split(/\s+/)[1].to_i * 1024 if line.match(/^Buffers/)
-      mem['cached']    = line.split(/\s+/)[1].to_i * 1024 if line.match(/^Cached/)
-      mem['swapTotal'] = line.split(/\s+/)[1].to_i * 1024 if line.match(/^SwapTotal/)
-      mem['swapFree']  = line.split(/\s+/)[1].to_i * 1024 if line.match(/^SwapFree/)
-      mem['dirty']     = line.split(/\s+/)[1].to_i * 1024 if line.match(/^Dirty/)
+      if line
+        mem['total']     = line.split(/\s+/)[1].to_i * 1024 if line.match(/^MemTotal/)
+        mem['free']      = line.split(/\s+/)[1].to_i * 1024 if line.match(/^MemFree/)
+        mem['buffers']   = line.split(/\s+/)[1].to_i * 1024 if line.match(/^Buffers/)
+        mem['cached']    = line.split(/\s+/)[1].to_i * 1024 if line.match(/^Cached/)
+        mem['swapTotal'] = line.split(/\s+/)[1].to_i * 1024 if line.match(/^SwapTotal/)
+        mem['swapFree']  = line.split(/\s+/)[1].to_i * 1024 if line.match(/^SwapFree/)
+        mem['dirty']     = line.split(/\s+/)[1].to_i * 1024 if line.match(/^Dirty/)
+      end 
```

Comment:
It doesn't seem like this actually helps here since the regex should fail for each of theses anyway?